File: idea-plugin/src/main/kotlin/com/lingce/aijanitor/core/AiClient.kt
package com.lingce.aijanitor.core

import com.google.gson.Gson
import com.google.gson.JsonObject
import com.google.gson.JsonParser
import com.intellij.openapi.diagnostic.logger
import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.time.Duration

/**
 * Minimal client for any OpenAI-compatible `/chat/completions` endpoint.
 */
class AiClient(
    private val baseUrl: String,
    private val apiKey: String,
    private val model: String,
) {
    private val gson = Gson()
    private val http: HttpClient = HttpClient.newBuilder()
        .connectTimeout(Duration.ofSeconds(20))
        .build()

    /** Sends a chat completion request and returns the assistant message content. */
    fun chat(systemPrompt: String, userPrompt: String): String {
        val body = JsonObject().apply {
            addProperty("model", model)
            addProperty("temperature", 0.0)
            add("messages", gson.toJsonTree(
                listOf(
                    mapOf("role" to "system", "content" to systemPrompt),
                    mapOf("role" to "user", "content" to userPrompt),
                )
            ))
        }

        val request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint()))
            .timeout(Duration.ofSeconds(120))
            .header("Content-Type", "application/json")
            .header("Authorization", "Bearer $apiKey")
            .POST(HttpRequest.BodyPublishers.ofString(gson.toJson(body)))
            .build()

        val response = http.send(request, HttpResponse.BodyHandlers.ofString())
        if (response.statusCode() / 100 != 2) {
            throw RuntimeException("AI 请求失败 (HTTP ${response.statusCode()}): ${response.body().take(500)}")
        }
        val json = JsonParser.parseString(response.body()).asJsonObject
        return json.getAsJsonArray("choices")
            .first().asJsonObject
            .getAsJsonObject("message")
            .get("content").asString
    }

    private fun endpoint(): String {
        val trimmed = baseUrl.trimEnd('/')
        return if (trimmed.endsWith("/chat/completions")) trimmed else "$trimmed/chat/completions"
    }

    companion object {
        private val LOG = logger<AiClient>()
    }
}

File: app/services/ai_client.py
from __future__ import annotations

import json
import os
from typing import Any

import httpx


class AIClient:
    def __init__(self) -> None:
        self.api_key = os.getenv("OPENAI_API_KEY", "")
        self.base_url = os.getenv("OPENAI_BASE_URL", "https://api.openai.com/v1")
        self.model = os.getenv("OPENAI_MODEL", "gpt-4o-mini")

    @property
    def enabled(self) -> bool:
        return bool(self.api_key)

    def generate_json(self, system_prompt: str, user_prompt: str) -> dict[str, Any] | None:
        if not self.enabled:
            return None

        url = f"{self.base_url.rstrip('/')}/chat/completions"
        headers = {"Authorization": f"Bearer {self.api_key}"}
        payload = {
            "model": self.model,
            "messages": [
                {"role": "system", "content": system_prompt},
                {"role": "user", "content": user_prompt},
            ],
            "temperature": 0.1,
            "response_format": {"type": "json_object"},
        }
        try:
            response = httpx.post(url, headers=headers, json=payload, timeout=35)
            response.raise_for_status()
            message = response.json()["choices"][0]["message"]["content"]
            return json.loads(message)
        except (httpx.HTTPError, KeyError, json.JSONDecodeError, ValueError):
            return None
